math/src/elliptic_curve/element.rs

```diff
@@ -0,0 +1,232 @@
+use crate::cyclic_group::IsCyclicGroup;
+use crate::elliptic_curve::traits::IsEllipticCurve;
+use crate::field::element::FieldElement;
+use std::fmt::Debug;
+
+/// Represents an elliptic curve point using the projective short Weierstrass form:
+/// y^2 * z = x^3 + a * x * z^2 + b * z^3,
+/// where `x`, `y` and `z` variables are field elements.
+#[derive(Debug, Clone)]
+pub struct EllipticCurveElement<E: IsEllipticCurve> {
+    pub value: [FieldElement<E::BaseField>; 3],
+}
+
+impl<E: IsEllipticCurve> EllipticCurveElement<E> {
+    /// Creates an elliptic curve point giving the projective [x: y: z] coordinates.
+    pub fn new(value: [FieldElement<E::BaseField>; 3]) -> Self {
+        assert_eq!(
+            E::defining_equation(&value),
+            FieldElement::zero(),
+            "Point ({:?}) does not belong to the elliptic curve.",
+            &value
+        );
+        Self { value }
+    }
```

Comment:
This function should return a `Result`instead of using an `assert` because doing the latter would panic in runtime. 

If you don't have any choice but to panic (this is not the case tho) it is better to describe the cases in which that function would panic and why.